src/plone/api/content.py

```diff
@@ -666,3 +668,52 @@ def find(context=None, depth=None, unrestricted=False, **kwargs):
         return catalog.unrestrictedSearchResults(**query)
     else:
         return catalog(**query)
+
+
+@required_parameters("obj")
+def get_parents(obj: None, predicate: None, interface: None):
```

Comment:
You're using type hints here but you should set default values. That's why the tests are failing.

```suggestion
def get_parents(obj=None, predicate=None, interface=None):
```

Reply:
done with changes, yet the tests are failing!

Reply:
I'm getting this:
```
Total: 478 tests, 4 failures, 4 errors and 1 skipped in 38.404 seconds.
test: exit 1 (40.95 seconds) /home/rohnsha0/code/plone/plone.api> zope-testrunner --all --test-path=/home/rohnsha0/code/plone/plone.api/src -s plone.api pid=4250
  test: FAIL code 1 (44.92=setup[3.97]+cmd[40.95] seconds)
  evaluation failed :( (45.14 seconds)
make: *** [Makefile:61: test] Error 1
```

by running `make test`

Reply:
but unable to understand what it says! @davisagli

Reply:
the issue is I am unable to understand what the error is, else would've easily fixed :(

Reply:
Looking at the `Makefile`, `make test` invokes `tox -e test` from the virtual environment, which in turn is configured in `tox.ini`. Relevant information is in this stanza:

https://github.com/plone/plone.api/blob/a876d70649458e6eecb6786156cc53eae4add5e9/tox.ini#L106-L141

My questions are:

1.  Should we use the constraints file `https://dist.plone.org/release/6.0-dev/constraints.txt`? This question is for the maintainers of `plone.api`.
2.  Try running the command `zope-testrunner --all --test-path={toxinidir}/src -s plone.api {posargs}`, resolving the slugs for your system accordingly. See its docs at https://zopetestrunner.readthedocs.io/en/latest/.

I'm sorry I am not much help, but that's how I would start to figure it out. I know nothing about `zope-testrunner` other than it does what its name says.

Reply:
> Try running the command zope-testrunner --all --test-path={toxinidir}/src -s plone.api {posargs}, resolving the slugs for your system accordingly. See its docs at https://zopetestrunner.readthedocs.io/en/latest/.

okay now it doesnt run any tests, strangely

Reply:
What command did you try? Is there any output?

I'm not sure where the slug `{posargs}` gets its key/value pair.

Also I see this in `tox.ini`:

```ini
set_env =
    ROBOT_BROWSER=headlesschrome
```

Which indicates you need to do the following before issuing the `zope-testrunner` command in the current terminal session.

```shell
export ROBOT_BROWSER=headlesschrome
```

Reply:
i already set the robot_browser to env

> zope-testrunner --all --test-path=src -s plone.api

Total: 0 tests, 0 failures, 0 errors and 0 skipped in 0.000 seconds.

Reply:
@stevepiercy You're sending @rohnsha0 on a bit of a wild goose chase. It's possible to run zope-testrunner directly but I don't think there's anything wrong using `make test` to run the tests. And these aren't Robot Framework tests so ROBOT_BROWSER is not relevant.

@rohnsha0 `4 failures, 4 errors` means you have to look further up in the output, find those specific tests that failed, study what went wrong, and adjust either the tests or the implementation to fix it. I saw at least some of the failing tests are in the documentation.

I can take a look myself and try to help, but it will probably take me some days to get to it, since I have a lot of things to review.

Reply:
> @rohnsha0 4 failures, 4 errors means you have to look further up in the output

I'm out of station since morning. Will be trying again tomorrow. However I've figured out the reason, some tests were failing in `test_content.py` and others in doc

Reply:
its all green now! thanks @stevepiercy  @davisagli